dali/kernels/imgproc/convolution/convolution_gpu.h

```diff
@@ -88,15 +115,21 @@ struct ConvolutionGpu {
   void Run(KernelContext& ctx, const TensorListView<StorageGPU, Out, ndim> out,
            const TensorListView<StorageGPU, const In, ndim>& in,
            const TensorListView<StorageCPU, const W, 1>& windows,
-           span<const int> window_anchors = {}, float scale = 1) {
+           const span<const int> window_anchors = {}, const ConvEpilogue& conv_epilogue = 1.f) {
     int num_samples = in.size();
+    int num_scales = conv_epilogue.num_samples();
 
     DALI_ENFORCE(
         window_anchors.size() == num_samples || window_anchors.size() == 0,
         make_string(
             "Unexpected number of window_anchors, expected either anchors for all samples ( ",
             num_samples,
             ") or no anchors for windows centered by default, got: ", window_anchors.size(), "."));
+
+    DALI_ENFORCE(num_scales == 1 || num_scales == num_samples,
+                 make_string("Scale argument must be either 1 or equal to batch size (",
+                             num_samples, "), got: ", num_scales, "."));
```

Comment:
This is wrong - if num_scales is 1, but the epilogue was constructed with a 1-element span, this is an error and it will result in an out-of-bounds read when accessing `conv_epilogue.alpha(i)`.

Reply:
Good point. I am not sure if it is really necessary to validate if one passed empty span of floats (as opposed to span being empty because of the scalar constructor used), because it is kind-of internal functionality, so I just default so scalar =1.f in that case.